raster-foundry/granary
# Kick off batch jobs when new predictions are created

## Overview

This PR attempts to kick off batch jobs configured for a model when new predictions are created. It is the first tame PR in granary history. Very exciting time to be alive.

### Checklist

- ~New tables and queries have appropriate indices added~
- ~Any new migrations have been audited to make sure they won't kill the application while being applied~
- ~Any new API endpoints have tests~
- ~Any frontend changes are hidden behind a feature flag~

## Notes

wip until I resolve the TODO in PredictionDao and test with the model we have up there

## Testing Instructions

- load development data if you haven't (oops)
- reassemble api server
- set ENVIRONMENT to `production` in docker-compose.yml for the `api` service
- `export AWS_PROFILE=raster-foundry`
- `./scripts/server`
- `echo '{"modelId": "b65116e0-60a5-4979-86bd-f40b54a1cb6f", "arguments": {"RED_BAND": "s3://foo/bar.tif", "GREEN_BAND": "s3://foo/bar.tif", "OUTPUT_LOCATION": "s3://foo/bar.geojson"}}' | http :8080/api/predictions`
- it should succeed and tell you that your job is started (it's never going to finish, sorry
- `./scripts/dbshell`, and:

```sql
insert into models select uuid_generate_v4(), 'water-bogus-job-def', validator, 'jobDefDoesNotExist', job_queue from models where id = 'b65116e0-60a5-4979-86bd-f40b54a1cb6f';
```

- get that model's id
- `echo '{"modelId": "<the bogus model's id>", "arguments": {"RED_BAND": "s3://foo/bar.tif", "GREEN_BAND": "s3://foo/bar.tif", "OUTPUT_LOCATION": "s3://foo/bar.geojson"}}' | http :8080/api/predictions`
- you should get a really surprisingly helpful error message about what's probably wrong

Closes #25
